event-schema/counter_manager.go

```diff
@@ -36,15 +44,49 @@ func getAllFrequencyCounters(schemaHash string) []*FrequencyCounter {
 	return frequencyCounters
 }
 
-func getFrequencyCounter(schemaHash, key string) *FrequencyCounter {
+// pruneFrequencyCounters brings the frequency counters back to desired bound.
+func pruneFrequencyCounters(schemaHash string, bound int) {
```

Comment:
Can we have a test to check that the prune happens as expected?

Reply:
I'll add that testcase as well. Thanks 👍🏼